### DISCUSSION TITLE
如何自定义绕过核心？

### DISCUSSION THREAD

**Discussion by cuminlo** [nikkinikki-org/OpenWrt-nikki#362] [Q&A] [answered]
目前提供的绕过大陆功能很好用，不用经过核心，小设备上效率提升还是比较明显。

现在我想要绕过一些自定的IP应该如何编写防火墙规则？规则写在哪个位置比较好？

**Comment by morytyann** [comment]
添加到china_ip这个集合即可

**Reply by cuminlo** [reply]
> 添加到china_ip这个集合即可

1. 是直接添加到 `geoip_cn.nft` 这个文件中么？这样不方便以后更新。
2. 还是使用 `nft add element inet mihomo china_ip { 2.56.0.0 }` 这样的添加？可是重启机器或者服务后还要执行一次，怎么能更加方便一点，大佬 @morytyann

**Reply by morytyann** [reply, marked as answer]
可以添加到文件，代价是更新或者重装就没了，通过命令也可以。
加到保留地址那里吧，效果一样，更新或者重装也不会丢。

**Reply by cuminlo** [reply]
> 可以添加到文件，代价是更新或者重装就没了，通过命令也可以。 加到保留地址那里吧，效果一样，更新或者重装也不会丢。

好的，我加到 `reserved_ip.nft` 文件里面先用一段时间，谢谢。
